lib/active_merchant/billing/gateways/worldpay.rb

```diff
@@ -1066,11 +1075,11 @@ def error_code_from(success, raw)
         raw[:iso8583_return_code_code] || raw[:error_code] || nil unless success == 'SUCCESS'
       end
 
-      def required_status_message(raw, success_criteria, action)
-        return if success_criteria.include?(raw[:last_event])
+      def required_status_message(raw, success_or_failure_criteria, action)
+        return if success_or_failure_criteria.include?(raw[:last_event])
         return unless %w[cancel refund inquiry credit fast_credit].include?(action)
 
-        "A transaction status of #{success_criteria.collect { |c| "'#{c}'" }.join(' or ')} is required."
+        "A transaction status of #{success_or_failure_criteria.collect { |c| "'#{c}'" }.join(' or ')} is required."
```

Comment:
```suggestion
        required_status = success_or_failure_criteria.map { |status| "'#{status}'" }.join(' or ')
        "A transaction status of #{required_status} is required."
```

Reply:
I wish I saw this earlier but I'm going to leave as is for now. Since I'm deploying soon